desktop_version/src/Render.cpp

```diff
@@ -2040,6 +2040,143 @@ static void draw_roomname_menu(void)
  * the same in Flip Mode. */
 #define FLIP(y, h) (graphics.flipmode ? 220 - (y) - (h) : (y))
 
+void rendermap(void)
+{
+#ifndef NO_CUSTOM_LEVELS
+    if (map.custommode)
+    {
+        //draw the map image
+        graphics.drawpixeltextbox(35 + map.custommmxoff, 16 + map.custommmyoff, map.custommmxsize + 10, map.custommmysize + 10, 65, 185, 207);
+        graphics.drawpartimage(graphics.minimap_mounted ? 1 : 12, 40 + map.custommmxoff, 21 + map.custommmyoff, map.custommmxsize, map.custommmysize);
+        return;
+     }
+#endif /* NO_CUSTOM_LEVELS */
+
+    //draw the map image
+    graphics.drawpixeltextbox(35, 16, 250, 190, 65, 185, 207);
+    graphics.drawimage(1, 40, 21, false);
+}
+
+void rendermapfog(void)
+{
+    int mapwidth = map.custommode ? map.customwidth : 20;
+    int mapheight = map.custommode ? map.customheight : 20;
+    int mapzoom = map.custommode ? map.customzoom : 1;
+    int mapxoff = map.custommode ? map.custommmxoff : 0;
+    int mapyoff = map.custommode ? map.custommmyoff : 0;
+
+    // Draw the fog of war
+    for (int j = 0; j < mapheight; j++)
+    {
+        for (int i = 0; i < mapwidth; i++)
```

Comment:
```suggestion
        for (int i = 0; i < mapwidth; ++i)
```

Reply:
What's wrong with using `i++`, and right with using `++i`, in a for loop? I'm much more used to seeing `i++` in general, and it's not good for consistency either because `i++` is also much more common in VVVVVV itself:

```
desktop_version/src$ egrep -r '; *[A-Za-z0-9_]+ *\+\+ *)' | wc -l
318
desktop_version/src$ egrep -r '; *\+\+ *[A-Za-z0-9_]+ *)' | wc -l
41
```

Reply:
I prefer the prefix operator because it definitively guarantees there's no temporary generated to hold the previous value as opposed to the postfix operator. Although, it probably doesn't matter much because compilers _should_ figure out that the previous value isn't used in these cases. I didn't consider that the postfix was in dominant usage.